src/python_testing/TC_DISHM_3_3.py

```diff
@@ -0,0 +1,174 @@
+
+        logging.info("StartUpMode: %s" % (startup_mode_dut))
+
+        asserts.assert_true(type_matches(startup_mode_dut, uint) or startup_mode_dut ==
+                            NullValue, "Startup mode value should be an integer value or null")
+
+        if startup_mode_dut == NullValue:
+
+            self.print_step(3, "Read SupportedModes attribute")
+            supported_modes_dut = await self.read_mode_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.SupportedModes)
+
+            logging.info("SupportedModes: %s" % (supported_modes_dut))
+
+            asserts.assert_greater_equal(len(supported_modes_dut), 2, "SupportedModes must have at least two entries!")
+
+            for m in supported_modes_dut:
+                new_start_up_mode_th = m.mode
+                break
+
+            self.print_step(4, "Write to the StartUpMode Attribute")
+
+            ret = await self.write_start_up_mode(newMode=new_start_up_mode_th)
+            # There is no recorded response returned from StartUpMode when writing to StartUpMode.
+            logging.info("Write StartUpMode Return: %s" % ret)
+
+        else:
+            new_start_up_mode_th = startup_mode_dut
+
+        self.print_step(5, "Read OnMode attribute")
+
+        old_on_mode_dut = await self.read_mode_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.OnMode)
+
+        logging.info("OnMode: %s" % (old_on_mode_dut))
+
+        #asserts.assert_true(type_matches(old_on_mode_dut, uint) , "On mode value should be an integer value")
+
+        if old_on_mode_dut == startup_mode_dut:
+
+            self.print_step(6, "Read SupportedModes attribute")
+            supported_modes_dut = await self.read_mode_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.SupportedModes)
+
+            logging.info("SupportedModes: %s" % (supported_modes_dut))
+
+            asserts.assert_greater_equal(len(supported_modes_dut), 2, "SupportedModes must have at least two entries!")
+
+            for m in supported_modes_dut:
+                if m.mode != startup_mode_dut:
+                    new_mode_th = m.mode
+                    break
+
+            self.print_step(7, "Write to OnMode attribute with NewMode set to %d" % (new_mode_th))
+
+            ret = await self.write_on_mode(newMode=new_mode_th)
+            asserts.assert_true(ret.status == CommonCodes.SUCCESS.value, "Changing the Onmode should succeed")
+
+        self.print_step(8, "Read OnMode attribute")
+
+        new_on_mode_dut = await self.read_mode_attribute_expect_success(endpoint=self.endpoint, attribute=attributes.OnMode)
+
+        logging.info("OnMode: %s" % (new_on_mode_dut))
+        # There is an issue with the test plan here as new_mode_th cannot be read if step 7 is not executed so I've changed
+        # from comparing against new_mode_th to comparing against startup_mode_dut instead to ensure that it runs.
+        # Will make issue
```

Comment:
Please link to the issue here (in the code and the PR comment) once it's created.